thebeetoken/beenest-web
# Use sticky positioning for Header

## Description
This is a proposed simplification of our `Header` to use the `sticky-top` class instead of `fixed-top`

With `fixed-top`, the element no longer influences subsequent flow, so we end up needing to add padding elsewhere any time a `Header` is used (or lose content behind it)

Using `sticky-top` comes out with a similar end result, but the element is treated as part of the flow of the document, so new content comes after the header, only disappearing behind it on scroll.

As an example of the benefit of this, our custom `Jumbotron` wrapper is removed, and `Home` is no longer dependent on padding specified by that custom `Jumbotron` to display correctly.

Note that sticky support is [not universal](https://caniuse.com/#search=sticky), although it is fairly broad among evergreen browsers. Tested in Chromium 40; the header scrolls with the rest of the page, but the layout is otherwise identical.

## How to Test
1. Go to `/work`
2. Verify that page displays as it does on `master`

Which devices did you test on?
- [x] Chromium 40 on Mac

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?
